pkg/sql/crdb_internal.go

```diff
@@ -2633,6 +2636,36 @@ func writeCreateTypeDescRow(
 		// Multi-region enums are created implicitly, so we don't have create
 		// statements for them.
 		return false, nil
+	case descpb.TypeDescriptor_COMPOSITE:
+		name, err := tree.NewUnresolvedObjectName(2, [3]string{typeDesc.GetName(), sc.GetName()}, 0)
+		if err != nil {
+			return false, err
+		}
+		composite := typeDesc.TypeDesc().Composite
+		typeList := make([]tree.CompositeTypeElem, len(composite.TupleContents))
+		for i := range composite.TupleContents {
+			if err := typedesc.EnsureTypeIsHydrated(
+				ctx, composite.TupleContents[i], resolver.(catalog.TypeDescriptorResolver),
+			); err != nil {
+				return false, err
+			}
```

Comment:
Just curious: is this hydration logic exercised at all now that we can't reference user-defined types within composite types?

Reply:
Agreed in this case.